glean_parser/util.py

```diff
@@ -136,14 +135,32 @@ def ensure_list(value):
     return value
 
 
+def to_camel_case(input, capitalize_first_letter):
+    """
+    Convert the value to camelCase.
+
+    This additionally replaces any '.' with '_'. The first letter is capitalized
+    depending on `capitalize_first_letter`.
+    """
+    sanitized_input = input.replace(".", "_").replace("-", "_")
+    # Filter out any empty token. This could happen due to leading '_' or
+    # consecutive '__'.
+    tokens = [s.capitalize() for s in sanitized_input.split("_") if len(s) != 0]
+    # If we're not meant to capitalize the first letter, then lowercase it.
+    if not capitalize_first_letter:
+        tokens[0] = tokens[0].lower()
+    # Finally join the tokens and capitalize.
+    return ''.join(tokens)
```

Comment:
We could, alternatively, take the regex approach used in inflection, which given how Python works is probably much faster (which would require adding a copyright notice about the borrowing of code, of course).

Reply:
I thought about doing that, but I opted for rewriting this given my poor knowledge of regexes and adding more specific test cases for our purpose. I wasn't really sure how to handle copyright as well :(

Moreover, the 'inflection' library does not properly handle strings such as `_test__me`, which are transformed to `_test_me`. I'm not sure how often we'd hit that case (if any time at all!), but if you have no strong opinion on this I'd leave it as it is if only to save us the copyright trouble :)